### ISSUE TITLE
authorization process failed

### ISSUE THREAD

**Issue by verajosemanuel** [alphapapa/pocket-reader.el#14]
installed ok, but when started the authorization process does not work.
I get this message when I opened the website with the provided link:

`"The link you followed here is no longer working. Would you like to start over at the home page?` "

**Comment by llcc** [comment]
same here.

**Comment by alphapapa** [comment]
I just tested it and it seems to work fine.  Please try again.  It was probably a transient error.

**Comment by loopified** [comment]
Getting the same error here. Have you guys managed to successfully authorize it?

**Comment by alphapapa** [comment]
I just tested again.  I get the error from the Pocket web site, yet **it actually works** and saves the access token.  Try the `pocket-reader` command again.

If that still doesn't work, **log out of Pocket, then retry authorizing.**

**Comment by loopified** [comment]
I have just done that and here it unfortunately doesn't save the access token. Then I have tried to log out of Pocket and retry authorizing, as you instructed, and it still doesn't work. After that I did `pocket-lib-reset-auth` and retried the steps above and it still fails.

**Comment by loopified** [comment]
After a failed attempt, if I try to call `pocket-reader` again, this is printed to my Messages buffer:

```
helm-M-x: Wrong number of arguments: (lambda (alist) "Create a hash table with initial values according to ALIST." (let ((h (ht-create))) (let ((--dolist-tail-- (reverse alist)) pair) (while --dolist-tail-- (setq pair (car --dolist-tail--)) (let ((key (car pair)) (value (cdr pair))) (ht-set! h key value)) (setq --dolist-tail-- (cdr --dolist-tail--))) (setq pair nil) h))), 2
```

**Comment by alphapapa** [comment]
> I have just done that and here it unfortunately doesn't save the access token. Then I have tried to log out of Pocket and retry authorizing, as you instructed, and it still doesn't work. After that I did pocket-lib-reset-auth and retried the steps above and it still fails.

I don't know what else to suggest, besides trying to reproduce in a clean Emacs configuration.  I removed my saved token file and tested it, and the Pocket site gave me the error, but then I ran the command again, and it worked, loaded all the items, and I saw that the token file was saved.

> After a failed attempt, if I try to call pocket-reader again, this is printed to my Messages buffer:

That error is puzzling.  It *might* indicate a problem with your Emacs configuration or packages.  Please try uninstalling and reinstalling the packages `pocket-reader` and `ht`.

**Comment by verajosemanuel** [comment]
Tried again after error in website auth I get:

```
Wrong type argument: sequencep, 403

helm-M-x: Opening output file: No such file or directory, c:/Users/jose/.cache/emacs-pocket-lib-token.json
```

**Comment by loopified** [comment]
>  Please try uninstalling and reinstalling the packages pocket-reader and ht.

I have removed and reinstalled those packages and still get the same behavior. :/

**Comment by alphapapa** [comment]
I just verified, again, that it works.  Here are the steps I followed:

1.  In a browser, visited `getpocket.com` and logged out of Pocket.
2.  Deleted existing `~/.cache/emacs-pocket-lib-token.json` file.
3.  `cd` to MELPA git repo directory.
4.  `rm -rf sandbox/; make sandbox INSTALL=pocket-reader`.
5.  After installation finishes, run `M-x pocket-reader RET`.
6.  Paste URL from clipboard into browser.
7.  Type username and password into Pocket page, click `Authorize`.
8.  Ignore the `Page Not Found` error on the next Pocket page.
9.  `M-x pocket-reader RET` again.
10.  All my Pocket items are now visible in Emacs.

**Comment by xuchunyang** [comment]
FWIW, I encountered the same error, then invoking `M-x pocket-reader` again does work for me.

**Comment by dorneanu** [comment]
> I just verified, again, that it works. Here are the steps I followed:
> 
>     1. In a browser, visited `getpocket.com` and logged out of Pocket.
> 
>     2. Deleted existing `~/.cache/emacs-pocket-lib-token.json` file.
> 
>     3. `cd` to MELPA git repo directory.
> 
>     4. `rm -rf sandbox/; make sandbox INSTALL=pocket-reader`.
> 
>     5. After installation finishes, run `M-x pocket-reader RET`.
> 
>     6. Paste URL from clipboard into browser.
> 
>     7. Type username and password into Pocket page, click `Authorize`.
> 
>     8. Ignore the `Page Not Found` error on the next Pocket page.
> 
>     9. `M-x pocket-reader RET` again.
> 
>     10. All my Pocket items are now visible in Emacs.

I did exactly the same thing but I still get:

```
Wrong number of arguments: (lambda (alist) "Create a hash table with initial values according to ALIST." (let ((h (ht-create))) (let ((--dolist-tail-- (reverse alist)) pair) (while --dolist-tail-- (setq pair (car --dolist-tail--)) (let ((key (car pair)) (value (cdr pair))) (ht-set! h key value)) (setq --dolist-tail-- (cdr --dolist-tail--)))\
 (setq pair nil) h))), 2
```

**Comment by alphapapa** [comment]
@dorneanu That error indicates a problem with your Emacs configuration's installed packages, not an error with this package.

**Comment by alphapapa** [comment]
The original problem reported in this issue is documented in the readme, so I'm closing this issue.  Other errors mentioned here are unrelated to it.

**Comment by iT-Boyer** [comment]
this issues，How to fix?
```
How Can I Help?
The link you followed here is no longer working. Would you like to start over at the [home page](https://getpocket.com/)?

If you think there is a problem, [please let us know](http://help.getpocket.com/).
```

**Comment by brdloush** [comment]
FYI: For anyone hitting `The link you followed here is no longer working.` even in 2025+.

With the first auth attempt, I got the same message. At first I haven't actually noticed it and tried running `pocket-reader` command again in emacs, receiving some weird auth http response error. Running that command over and over again did not help.

Then I managed to read in readme.org that it's simply supposed to work and that you should try to run the command despite seeing the mentioned `The link you followed here is no longer working.` message.  So I gave my emacs one more restart, `pocket-reader` asked for yet another visit of pocket auth flow.. and this time, after that message, second call to `pocket-reader` brought a successful pocket reader screen with my saves being shown.

So basically.. if it doesn't work for you on your first attempt, try again. Maybe it will work on 2nd attempt 🤞. It did for me.
